types/index.d.ts

```diff
@@ -21,6 +22,52 @@ export interface LogFacesHTTPAppender {
    * Make sure you use the correct type base on your url
    */
   agent?: httpAgent | httpsAgent;
+  /** Adjust the resulting logfacesEvent that is sent out.
+   *
+   * Needs to return the new layout or undefined to ignore the event.
+   *
+   * If `LogFacesEvent.m` is nullish: it will be populated with the default formatter
+   * */
+  eventLayout?: LogFacesLayoutFunction;
+}
+
+export type LogFacesLayoutFunction = (
+  loggingEvent: LoggingEvent,
+  logFacesEvent: Omit<LogFacesEvent, 'm'>
+) => LogFacesEvent | undefined | null;
+
+/** [Data model: Log events](http://www.moonlit-software.com/logfaces/downloads/logfaces-manual.pdf) */
+export interface LogFacesEvent {
+  /** Time stamp as specified by the source or server */
+  t: number;
+  /** Severity of event expressed in term of log4j levels */
+  p: number | string;
+  /**  Name of the domain (or application) originating the event */
+  a: string;
+  /** Name of the host originating the event */
+  h?: string;
+  /** Name of the logger (class, module, etc) originating the event */
+  g: string;
+  /** Name of the thread originating the event */
+  r?: string;
+  /** Message Content */
+  m?: string;
+  /** [Network Diagnostic Context](http://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/NDC.html) */
+  n?: unknown;
```

Comment:
According to page 18 of [logfaces-manual.pdf](http://www.moonlit-software.com/logfaces/downloads/), it seems to take in Java `String`.
For us, should it be `string`, `unknown` or `any`?